aptos-move/aptos-vm-environment/src/environment.rs

```diff
@@ -297,5 +302,86 @@ pub mod test {
         );
     }
 
-    // TODO(loader_v2): Add equality tests.
+    fn state_view_with_non_default_config<T: OnChainConfig + Serialize>(
+        config: T,
+    ) -> MockStateView<StateKey> {
+        MockStateView::new(HashMap::from([(
+            StateKey::resource(T::address(), &T::struct_tag()).unwrap(),
+            StateValue::new_legacy(bcs::to_bytes(&config).unwrap().into()),
+        )]))
+    }
+
+    #[test]
+    fn test_environment_eq() {
+        let state_view = MockStateView::empty();
+        let environment_1 = AptosEnvironment::new(&state_view);
+        let environment_2 = AptosEnvironment::new(&state_view);
+        assert!(environment_1 == environment_2);
+    }
+
+    #[test]
+    fn test_environment_ne_configuration() {
+        let mut configuration = ConfigurationResource::default();
+        assert_eq!(configuration.last_reconfiguration_time_micros(), 0);
+        configuration.set_last_reconfiguration_time_for_test(1);
+
+        let environment_1 = AptosEnvironment::new(&MockStateView::empty());
+        let environment_2 =
+            AptosEnvironment::new(&state_view_with_non_default_config(configuration));
+        assert!(environment_1 != environment_2);
+    }
+
+    #[test]
+    fn test_environment_ne_chain_id() {
```

Comment:
you could combine with the above and check ne among all 3 pairs. 
just saw you do similar for gas versions, wonder if it makes sense to check ne between a larger set of environments instead of separate tests, but up to you.